airbyte-workers/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -308,6 +324,29 @@ else if (hasFailed.get()) {
 
   }
 
+  private class ThreadedTimeHolder {
+
+    private long sourceReadEndTime = -1;
+    private long destinationWriteEndTime = -1;
+
+    public void trackSourceReadEndTime() {
+      this.sourceReadEndTime = System.currentTimeMillis();
+    }
+
+    public void trackDestinationWriteEndTime() {
+      this.destinationWriteEndTime = System.currentTimeMillis();
+    }
+
+    public long getSourceReadEndTime() {
+      return this.sourceReadEndTime;
+    }
+
+    public long getDestinationWriteEndTime() {
+      return this.destinationWriteEndTime;
+    }
+
+  }
```

Comment:
This probably isn't the right way to create a thread-safe object in Java for extracting local information from our Runnable threads below.  Any tips?

Reply:
possibly this? https://www.baeldung.com/java-synchronized

Reply:
Nide! Synchronized via 0a8e619116914736377318ab1d68f456ab7ec1f1